### DISCUSSION TITLE
考虑将bulidKey中的判断逻辑下放到KeyConvertor中吗？

### DISCUSSION THREAD

**Discussion by gaoying0805** [alibaba/jetcache#752] [Q&A]
public byte[] buildKey(K key) {
        try {
            Object newKey = key;
            if (key instanceof byte[]) {
                newKey = key;
            } else if(key instanceof String){
                newKey = key;
            } else {
                if (config.getKeyConvertor() != null) {
                    newKey = config.getKeyConvertor().apply(key);
                }
            }
            return ExternalKeyUtil.buildKeyAfterConvert(newKey, config.getKeyPrefix());
        } catch (IOException e) {
            throw new CacheException(e);
        }
    }
考虑后续将key逻辑判断下放到具体的keyConvertor中吗，这样自定义拼接一类的会更加方便，我这面遇到了自定义后缀拼接时，key instanceof String 状态时拼接失效的问题。

**Comment by areyouok** [comment]
see https://github.com/alibaba/jetcache/issues/749
